### ISSUE TITLE
Implement Error Middleware for API

### ISSUE THREAD

**Issue by karankoder** [GauravGupta993/MERN-Templates#16]
I want to add error middleware to the server-side code and remove unnecessary code for better coding practices.

**Comment by karankoder** [comment]
@GauravGupta993 please review this issue.

**Comment by GauravGupta993** [comment]
@karankoder 
Yes, you may work on it. Also since this issue doesn't have much work, it would be great if you could additionally add the comments in the codebase wherever they are required or current comments seem lacking.
